examples/abitest/module_1/rust/src/lib.rs

```diff
@@ -25,18 +24,26 @@ use protobuf::ProtobufEnum;
 // response is written to a fresh channel, and the read handle for that new channel
 // is what's sent back to the frontend.
 #[no_mangle]
-pub extern "C" fn backend_oak_main(handle: u64) -> i32 {
-    std::panic::catch_unwind(|| {
+pub extern "C" fn backend_oak_main(handle: u64) {
+    let _ = std::panic::catch_unwind(|| {
         oak::set_panic_hook();
         main(handle)
-    })
-    .unwrap_or(Err(oak::OakStatus::ERR_INTERNAL))
-    .err()
-    .unwrap_or(oak::OakStatus::OK)
-    .value()
+    });
 }
 
-pub fn main(in_handle: u64) -> Result<(), oak::OakStatus> {
+pub fn main(in_handle: u64) {
+    match inner_main(in_handle) {
+        Err(oak::OakStatus::ERR_TERMINATED) => {
+            info!("node terminated");
+        }
```

Comment:
Personally I think we should get rid of `ERR_TERMINATED` entirely at some point, and reserve `OakStatus` as the return type of ABI function calls. Then I would change the type of `inner_main` to `-> Result<(), Box<dyn Debug>>` or similar, so that arbitrary errors may be handled with `?` throughout the `main` body, and the resulting error may be logged.

This is also similar to how it is allowed to have a [`Result` return value](https://github.com/rust-lang/rust/blob/master/RELEASES.md#language-13) from `main` in standard Rust.

WDYT?

Reply:
`ERR_TERMINATED` *is* a return type from ABI function calls, particularly `wait_on_channels` &ndash; it indicates that the Node should terminate.  Being propagated out as a return value from `oak_main` was mostly a side effect.